discovery/openstack/hypervisor_test.go

```diff
@@ -89,8 +85,6 @@ func TestOpenstackSDHypervisorRefresh(t *testing.T) {
 	} {
 		testutil.Equals(t, model.LabelValue(v), tg.Targets[1][model.LabelName(l)])
 	}
-
-	mock.TearDownSuite()
```

Comment:
This doesn't seem to be doing quite the same thing, as this is for the mock created in this function as against s.Mock.

Reply:
mock = s

Reply:
mock is  &OpenstackSDHypervisorTestSuite{}

func (s *OpenstackSDInstanceTestSuite) TearDownSuite()  is what I remove. That calls s.Mock.ShutdownServer().

Reply:
OpenstackSDInstanceTestSuite and OpenstackSDInstanceTestSuite both call mock.Setup

Reply:
The mock behind both is the same. I think this PR is correct.